scripts/build-and-test.sh

```diff
@@ -19,10 +19,25 @@ function run_tests_with_gdb {
   for x in bin/**/*-tests; do scripts/run-with-gdb.sh "$x"; done
 }
 
+function run_tests {
+    for x in bin/**/*-tests; do "$x"; done
+}
+
+num_procs=1
+if [[ "$(uname)" == "Darwin" ]]; then
+    num_procs=$(sysctl -n hw.ncpu)
```

Comment:
We'll want `ysctl -n hw.physicalcpu` to get the `physical cores` on OSX it seems :)